lib/service/discovery.go

```diff
@@ -26,7 +26,8 @@ import (
 )
 
 func (process *TeleportProcess) shouldInitDiscovery() bool {
-	return process.Config.Discovery.Enabled && (len(process.Config.Discovery.AWSMatchers) != 0 || len(process.Config.Discovery.AzureMatchers) != 0)
+	return process.Config.Discovery.Enabled && (len(process.Config.Discovery.AWSMatchers) != 0 ||
+		len(process.Config.Discovery.AzureMatchers) != 0 || len(process.Config.Discovery.GCPMatchers) != 0)
```

Comment:
Nit: Now about we make a helper `IsEmpty() bool` method on `process.Config.Discovery` and use it here?